Sources/Jetworking/Client/Client.swift

```diff
@@ -197,15 +221,19 @@ public final class Client {
     }
 
     @discardableResult
-    public func delete<ResponseType>(endpoint: Endpoint<ResponseType>, parameter: [String: Any] = [:], _ completion: @escaping RequestCompletion<ResponseType>) -> CancellableRequest? {
+    public func delete<ResponseType: Decodable>(endpoint: Endpoint<ResponseType>, parameter: [String: Any] = [:], _ completion: @escaping RequestCompletion<ResponseType>) -> CancellableRequest? {
         do {
             let request: URLRequest = try createRequest(forHttpMethod: .DELETE, and: endpoint)
             return requestExecutor.send(request: request) { [weak self] data, urlResponse, error in
-                self?.handleResponse(
+                guard let self = self else { return }
+
+                ResponseHandler.evaluate(
                     data: data,
                     urlResponse: urlResponse,
                     error: error,
                     endpoint: endpoint,
+                    configuration: self.configuration,
+                    completionWrapper: ResponseHandler.CompletionWrapper.decodable,
                     completion: completion
```

Comment:
Can't we pass the wrapped completion instead of the wrapper and the completion?

Reply:
Only if we duplicate the `evaluate` method: One with `evaluate<ResponseType>` and `evaluate<ResponseType: Decodable>`. The current approach was chosen to prevent this code duplication.

Reply:
Hm what do you think if we create two functions as you say but just pass the completion wrapper there? Then we could handle it directly in the `ResponseHandler` for example in following way:

```swift
func handleVoidResponse<ResponseType>(....) {
    handleResponse(
        ...
        completionWrapper: void,
        ...
     )
}

func handleDecodableResponse<ResponseType: Decodable>(){
    handleResponse(
        ...
        completionWrapper: decodable,
        ...
     )
}

private func evaluate<ResponseType>(
         data: Data?,
         urlResponse: URLResponse?,
         error: Error?,
         endpoint: Endpoint<ResponseType>? = nil,
         configuration: Configuration,
         completionWrapper: @escaping WrappedCompletion<ResponseType>,
         completion: @escaping Client.RequestCompletion<ResponseType>
     ) { }
     ```

Reply:
What I'm trying to do with this approach ist to reduce the complexity of one call by reducing the parameters.

Reply:
The `completionWrapper` parameter is now removed and the selection is handled by methods in `ResponseHandler`